internal/inventory/azurefetcher/fetcher_resource_graph.go

```diff
@@ -0,0 +1,103 @@
+
+package azurefetcher
+
+import (
+	"context"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+
+	"github.com/elastic/cloudbeat/internal/inventory"
+	azurelib "github.com/elastic/cloudbeat/internal/resources/providers/azurelib/inventory"
+)
+
+type resourceGraphFetcher struct {
+	logger   *logp.Logger
+	provider resourceGraphProvider
+}
+
+type (
+	resourceGraphProvider interface {
+		ListAllAssetTypesByName(ctx context.Context, assetGroup string, assets []string) ([]azurelib.AzureAsset, error)
+	}
+)
+
+func newResourceGraphFetcher(logger *logp.Logger, provider resourceGraphProvider) inventory.AssetFetcher {
+	return &resourceGraphFetcher{
+		logger:   logger,
+		provider: provider,
+	}
+}
+
+func (f *resourceGraphFetcher) Fetch(ctx context.Context, assetChan chan<- inventory.AssetEvent) {
+	resourcesToFetch := []struct {
+		name           string
+		azureGroup     string
+		azureType      string
+		classification inventory.AssetClassification
+	}{
+		{"App Services", azurelib.AssetGroupResources, azurelib.WebsitesAssetType, inventory.AssetClassificationAzureAppService},
+		{"Container Registries", azurelib.AssetGroupResources, azurelib.ContainerRegistryAssetType, inventory.AssetClassificationAzureContainerRegistry},
+		{"Cosmos DB Accounts", azurelib.AssetGroupResources, azurelib.DocumentDBDatabaseAccountAssetType, inventory.AssetClassificationAzureCosmosDBAccount},
+		{"Cosmos DB SQL Databases", azurelib.AssetGroupResources, azurelib.CosmosDBForSQLDatabaseAssetType, inventory.AssetClassificationAzureCosmosDBSQLDatabase},
+		{"Disks", azurelib.AssetGroupResources, azurelib.DiskAssetType, inventory.AssetClassificationAzureDisk},
+		{"Elastic Pools", azurelib.AssetGroupResources, azurelib.ElasticPoolAssetType, inventory.AssetClassificationAzureElasticPool},
+		{"MySQL Flexible Servers", azurelib.AssetGroupResources, azurelib.FlexibleMySQLDBAssetType, inventory.AssetClassificationAzureSQLServer},
+		{"Resource Groups", azurelib.AssetGroupResourceContainers, azurelib.ResouceGroupAssetType, inventory.AssetClassificationAzureResourceGroup},
+		{"SQL Databases", azurelib.AssetGroupResources, azurelib.MySQLDatabaseAssetType, inventory.AssetClassificationAzureSQLDatabase},
+		{"Snapshots", azurelib.AssetGroupResources, azurelib.SnapshotAssetType, inventory.AssetClassificationAzureSnapshot},
+		{"Storage Accounts", azurelib.AssetGroupResources, azurelib.StorageAccountAssetType, inventory.AssetClassificationAzureStorageAccount},
+		{"Virtual Machines", azurelib.AssetGroupResources, azurelib.VirtualMachineAssetType, inventory.AssetClassificationAzureVirtualMachine},
+	}
+	for _, r := range resourcesToFetch {
+		f.fetch(ctx, r.name, r.azureGroup, r.azureType, r.classification, assetChan)
+	}
+}
+
+func (f *resourceGraphFetcher) fetch(ctx context.Context, resourceName, resourceGroup, resourceType string, classification inventory.AssetClassification, assetChan chan<- inventory.AssetEvent) {
+	f.logger.Infof("Fetching %s", resourceName)
```

Comment:
seems like we could remove `resourceName` and use `classification.subType` instead

Reply:
Sure, good idea :)